### PR TITLE
7章 unwinding.md を翻訳

### PR DESCRIPTION
**Pull request by joemphilips**

### CODE REVIEW

File: src/unwinding.md [hunk 1]
```diff
@@ -1,36 +1,77 @@
...
+裁量次第でpanicやabortさせることも可能だからです。panicはスレッドの正常処理を
+停止し、stackをunwind、全ての関数が即座にreturnしたかのようにデストラクタ
+を呼び出します。
 
+<!--
 As of 1.0, Rust is of two minds when it comes to panics. In the long-long-ago,
 Rust was much more like Erlang. Like Erlang, Rust had lightweight tasks,
 and tasks were intended to kill themselves with a panic when they reached an
-untenable state. Unlike an exception in Java or C++, a panic could not be
+untenable state. Unlike an exception in Java or C++, a panic could not bek
```
Comment (imsut) [inline src/unwinding.md]: can you revert this (seemingly unintentional) change?

File: src/exception-safety.md [hunk 2]
```diff
@@ -55,19 +96,33 @@ impl<T: Clone> Vec<T> {
...
 Vec that we definitely know has capacity. The logic is totally correct, except
 there's a subtle problem with our code: it's not exception-safe! `set_len`,
 `offset`, and `write` are all fine; `clone` is the panic bomb we over-looked.
-
+-->
+余計なキャパシティの確保と、確実にキャパシティを持つとわかっているVecに対する
+`len`チェックを避けるため、`push`の使用を避けています。このコードはロジックは
+完全に正しいのですが、ちょっとした問題があります。例外安全ではないのです！
+`set_len`、`offset`、`write`は全く問題がないのですが、`clone`がpanicの火種
+となります。
```
Comment (imsut) [inline src/exception-safety.md]: 「・・の火種となることを見落としていました。」
ではどうでしょう?

File: src/exception-safety.md [hunk 3]
```diff
@@ -55,19 +96,33 @@ impl<T: Clone> Vec<T> {
...
+Cloneは完全に我々の制御から完全に外れており、勝手にpanicする可能性がります。
+そうなった場合、Vecの長さが長すぎる状態になったまま、関数がexitしますその後に
+Vecが使われたりdropされたら、未初期化のメモリが読まれることになるのです！
 
+<!--
 The fix in this case is fairly simple. If we want to guarantee that the values
 we *did* clone are dropped, we can set the `len` every loop iteration. If we
 just want to guarantee that uninitialized memory can't be observed, we can set
 the `len` after the loop.
+-->
```
Comment (imsut) [inline src/exception-safety.md]: looks like this is work in progress?

**Comment by joemphilips** [comment]
> can you separate src/exception-safety.md out of this request so that you can merge unwinding first?

I was intending to translate all `Unwinding` section as single PR, but sent this PR before finishing my job so that others do not do the same job at the same time. ( That was the intension behind the `WIP:` )
but OK, I will send as different PR then.

**Comment by joemphilips** [comment]
I've finished translating `unwinding.md` , the reason I haven't change the `SUMMARY.md` is because I thought it is better to leave untranslated for the reader's sake

File: src/exception-safety.md (hunk 3 above)
Comment (joemphilips) [reply src/exception-safety.md]: yep, #6

**Pull request merged**